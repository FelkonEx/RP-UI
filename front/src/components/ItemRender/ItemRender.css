.item-render-container {
  margin-left: 1rem;
  margin-right: 1rem;
  margin-bottom: 1rem;

  img {
    border: 0.25rem solid #00BCD4;

    border-radius: 1rem;
  }
}

.item-info-headings {
  display: flex;
  justify-content: space-between;
  font-size: 2rem;
  margin-bottom: 1rem;

  .left {
    font-weight: bold;
    text-align: left;
  }

  .right {
    font-style: italic;
    text-align: right;
    font-size: 1.5rem;
  }
}

.item-info-date {
  margin-bottom: 0.5rem;
  font-size: 1.25rem;
}
