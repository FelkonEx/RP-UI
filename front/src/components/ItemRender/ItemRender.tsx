import "./ItemRender.css";

type ItemRenderProps = {
  heroImage: string | undefined;
  unitName: string | undefined;
  unitStyleName: string | undefined;
  checkInDate: string | undefined;
};

const IMAGE_HEIGHT: number = 360;
const IMAGE_WIDTH: number = 480;

const generateImageURL = (heroImage?: string) => {
  if (!heroImage) {
    return "";
  }

  return (
    "https://cms.inspirato.com/ImageGen.ashx?image=" +
    heroImage +
    "&height=" +
    IMAGE_HEIGHT +
    "&width=" +
    IMAGE_WIDTH
  );
};

const renderDate = (date: string) => {
  if (!date) {
    return "";
  }
  const dateObject = new Date(date);
  return `${dateObject.getDate()} - ${dateObject.getMonth()} - ${dateObject.getFullYear()}`;
};

export default function ItemRender({
  heroImage,
  unitName,
  unitStyleName,
  checkInDate,
}: ItemRenderProps) {
  return (
    <div
      className="item-render-container"
      style={{
        width: IMAGE_WIDTH,
      }}
    >
      <img src={generateImageURL(heroImage)} alt={unitName} />
      <div className="item-info-headings">
        {unitName && <span className="left">{unitName}</span>}
        <br />
        {unitStyleName && <span className="right">Style: {unitStyleName}</span>}
      </div>
      <div className="item-info-date">
        {checkInDate && (
          <span>Available to Check In: {renderDate(checkInDate)}</span>
        )}
      </div>
      <br />
    </div>
  );
}
