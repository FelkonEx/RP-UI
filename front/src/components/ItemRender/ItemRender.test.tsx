import { act } from "react"; // This should be from 'react' now
import { render } from "@testing-library/react";
import ItemRender from "./ItemRender";

it("renders correctly and matches snapshot", async () => {
  const { asFragment } = render(
    <ItemRender
      heroImage="test_hero_image.png"
      unitName="testUnitName"
      unitStyleName="testUnitStyleName"
      checkInDate="2024-01-06"
    />
  );

  /* eslint-disable testing-library/no-unnecessary-act */
  await act(async () => {});
  // from the console: `ReactDOMTestUtils.act` is deprecated in favor of `React.act`... but it's importing from react ðŸ¤”

  expect(asFragment()).toMatchSnapshot();
});
