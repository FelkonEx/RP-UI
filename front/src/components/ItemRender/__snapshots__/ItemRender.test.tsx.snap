// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="item-render-container"
    style="width: 480px;"
  >
    <img
      alt="testUnitName"
      src="https://cms.inspirato.com/ImageGen.ashx?image=test_hero_image.png&height=360&width=480"
    />
    <div
      class="item-info-headings"
    >
      <span
        class="left"
      >
        testUnitName
      </span>
      <br />
      <span
        class="right"
      >
        Style: testUnitStyleName
      </span>
    </div>
    <div
      class="item-info-date"
    >
      <span>
        Available to Check In: 6 / 1 / 2024
      </span>
    </div>
    <br />
  </div>
</DocumentFragment>
`;
